pinot-integration-tests/src/test/java/com/linkedin/pinot/integration/tests/QueryGenerator.java

```diff
@@ -15,103 +15,147 @@
  */
 package com.linkedin.pinot.integration.tests;
 
-import com.linkedin.pinot.common.utils.StringUtil;
+import java.io.BufferedWriter;
 import java.io.File;
+import java.io.FileWriter;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.HashSet;
+import java.util.Iterator;
 import java.util.List;
 import java.util.Map;
 import java.util.Random;
 import java.util.Set;
-import java.util.SortedSet;
-import java.util.TreeSet;
 import org.apache.avro.Schema;
 import org.apache.avro.file.DataFileReader;
 import org.apache.avro.generic.GenericData;
 import org.apache.avro.generic.GenericDatumReader;
 import org.apache.avro.generic.GenericRecord;
-import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.json.JSONArray;
+import org.json.JSONObject;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 
 /**
- * Utility class to generate random SQL queries based on an Avro file.
- *
+ * The <code>QueryGenerator</code> class is used to generate random equivalent PQL/SQL query pairs based on Avro files.
+ * <ul>
+ *   <li>
+ *     Supports COMPARISON, IN and BETWEEN predicate for both single-value and multi-value columns.
+ *     <ul>
+ *       <li>For multi-value columns, does not support NOT EQUAL and NOT IN.</li>
+ *       <li>The reason for this restriction is that number of elements in multi-value columns is not fixed.</li>
+ *     </ul>
+ *   </li>
+ *   <li>Supports single-value data type: BOOLEAN, INT, LONG, FLOAT, DOUBLE, STRING.</li>
+ *   <li>Supports multi-value data type: INT, LONG, FLOAT, DOUBLE, STRING.</li>
+ *   <li>
+ *     Supports aggregation function: SUM, MIN, MAX, AVG, COUNT, DISTINCTCOUNT.
+ *     <ul>
+ *       <li>SUM, MIN, MAX, AVG can only work on numeric single-value columns.</li>
+ *       <li>COUNT, DISTINCTCOUNT can work on any single-value columns.</li>
+ *     </ul>
+ *   </li>
+ * </ul>
  */
 public class QueryGenerator {
```

Comment:
We don't have a general purpose query generator yet (this one is the general one). Right now in  pinot-tools we only have a startree query generator (which only generate SUM aggregation queries). In the future we should move this into tools package.